src/test/c_api_stress_tests.cpp

```diff
@@ -0,0 +1,441 @@
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::defaultVersionAdd,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, KFSAddNewVersionDuringPredictLoad) {
+    bool performWholeConfigReload = false;                        // we just need to have all model versions rechecked
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        // XYZ &StressCapiConfigChanges::triggerPredictInALoop<TFSPredictRequest, TFSPredictResponse>,
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::defaultVersionAdd,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, GetMetricsDuringLoad) {
+    bool performWholeConfigReload = false;                        // we just need to have all model versions rechecked
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {};
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::testCurrentRequestsMetric,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, RemoveDefaultVersionDuringPredictLoad) {
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK,
+        StatusCode::PIPELINE_DEFINITION_NOT_LOADED_YET,  // we hit default version which is unloaded already but default is not changed yet
+        StatusCode::MODEL_VERSION_MISSING};              // there is no default version since all are either not loaded properly or retired
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE};
+    // we need whole config reload since there is no other way to dispose
+    // all model versions different than removing model from config
+    bool performWholeConfigReload = true;
+    performStressTest(
+        &StressCapiConfigChanges::triggerCApiInferenceInALoop,
+        &StressCapiConfigChanges::defaultVersionRemove,
+        performWholeConfigReload,
+        requiredLoadResults,
+        allowedLoadResults);
+}
+TEST_F(StressCapiConfigChanges, ChangeToShapeAutoDuringPredictLoad) {
+    bool performWholeConfigReload = true;
+    std::set<StatusCode> requiredLoadResults = {StatusCode::OK};  // we expect full continuity of operation
+    std::set<StatusCode> allowedLoadResults = {StatusCode::MODEL_NAME_MISSING};
```

Comment:
Here why we have model missing?
SHouldn't it be PIPELINE_DEFINITION_NOT_LOADED_YET and acutally as required LoadResult?

We make model unusable by the DAG, so the DAG should be stuck in loading error.
MODEL_NAME_MISSING suggests that it never existed in the config json.